### PR TITLE
Add attachment handling

### LINKED ISSUE #11: Attachment handling
We need to send attachments. A new parameter table would be comfy. Do you have any plans for that?
Or maybe the attachment could be put in the body concatenated. Do you have any examples for that?
Thx.

### PR DESCRIPTION
**Pull request by no1seman**
Add attachments handling api to lib. Full description in README.md and also some usefull comments in init.lua

Closes https://github.com/tarantool/smtp/issues/11

### CODE REVIEW

File: smtp/init.lua [hunk 1]
```diff
@@ -29,8 +29,8 @@
 --  SUCH DAMAGE.
 --
 
-local fiber = require('fiber')
```
Comment (Totktonada) [inline smtp/init.lua:32]: I still think that all those deletions contradict with the approach, when we trying to keep commits atomic. It is easy to extract them into its own commit using `git add -p`. However, okay, up to you.

File: smtp/init.lua [hunk 2]
```diff
@@ -155,27 +162,65 @@ curl_mt = {
...
             end
+            if opts.subject then
+                header = header .. 'Subject: ' .. opts.subject .. '\r\n'
+            end
             add_recipients(recipients, opts.bcc)
             if opts.headers and #opts.headers > 0 then
                 header = header .. table.concat(opts.headers, '\r\n') .. '\r\n'
             end
+            local content_type = 'Content-Type: ' .. (opts.content_type or 'text/plain') ..
+            '; charset=' .. (opts.charset or  'UTF-8') ..';\r\n'
```
Comment (Totktonada) [inline smtp/init.lua:181]: Nit: This header is added only when there are no attachments, so I would sink it down to the corresponding if-branch.

File: smtp/init.lua [hunk 3]
```diff
@@ -155,27 +162,65 @@ curl_mt = {
...
+                local MULTIPART_CONTENT_TYPE = 'Content-Type: multipart/mixed; boundary=MULTIPART-MIXED-BOUNDARY;\r\n'
+                local MULTIPART_SEPARATOR = '\r\n--MULTIPART-MIXED-BOUNDARY\r\n'
+                local MULTIPART_END = '\r\n--MULTIPART-MIXED-BOUNDARY--\r\n'
+                local attachments = ''
 
-            local content_type = opts.content_type or 'text/plain'
-            local charset = opts.charset or 'UTF-8'
-            header = header .. 'Content-Type: ' .. content_type .. '; charset=' .. charset .. '\r\n'
+                for _, attachment in ipairs(opts.attachments) do
+                    local base64_encode = attachment.base64_encode or true
```
Comment (Totktonada) [inline smtp/init.lua]: `false or true` is `true`.

File: smtp/init.lua [hunk 4]
```diff
@@ -155,27 +162,65 @@ curl_mt = {
...
+                    local attachment_content_type = 'Content-Type: ' ..
+                                         (attachment.content_type or 'text/plain') ..
+                                         '; charset=' ..
+                                         (attachment.charset or  'UTF-8') ..
+                                         ';\r\n'
+                    local content_transfer_encoding = attachment.base64_encode
```
Comment (Totktonada) [inline smtp/init.lua:199]: We can use `base64_encode` variable here.

**Review by Totktonada — COMMENTED** [review]
I'm okay in general, but there are several comments around. Most important: let's look at the case when `base64_encode` is set to `false`. It is invalid now: encoding is still performed, only `Content-Transfer-Encoding` header is omitted.

File: smtp/init.lua (hunk 2 above)
Comment (no1seman) [reply smtp/init.lua:181]: I'm disagree with your point cause 'content_type' is alo used in "with attachments" clause of if, see: lines 216-223 of init.lua
Generally in case of multipart bode content we still need point body content-type as it is a part of of multipart and every single part have to have content_type.

File: smtp/init.lua (hunk 3 above)
Comment (no1seman) [reply smtp/init.lua]: Fixed. Also I improved test suite to test both three cases: base64_encode = nil, false, true

File: smtp/init.lua (hunk 2 above)
Comment (Totktonada) [reply smtp/init.lua:181]: I overlooked it. Agreed.

**Comment by Totktonada** [comment]
Squashed commits, added `Closes #11`.

**Review by Totktonada — APPROVED** [review]
LGTM.
